File: src/model/article-data.ts
export interface AuthorData {
  imageUrl: string;
  username: string;
  following: boolean;
}

export interface ArticleData {
  favorited: boolean;
  author: AuthorData;
  tagList: string[];
  title: string;
  description: string;
  createdAt: string;
  favoritesCount: number;
  slug: string;
}

export interface CommentData {
  id: number;
  createdAt: Date;
  updatedAt: Date;
  body: string;
  author: AuthorData;
}

File: src/main/scala/com/softwaremill/realworld/articles/ArticleData.scala
package com.softwaremill.realworld.articles

import zio.json.{DeriveJsonDecoder, DeriveJsonEncoder, JsonDecoder, JsonEncoder}

import java.time.Instant

case class ArticleData(
    slug: String,
    title: String,
    description: String,
    body: String,
    tagList: List[String],
    createdAt: Instant,
    updatedAt: Instant,
    favorited: Boolean,
    favoritesCount: Int,
    author: ArticleAuthor
)

object ArticleData:
  given articleDataEncoder: JsonEncoder[ArticleData] = DeriveJsonEncoder.gen[ArticleData]
  given articleDataDecoder: JsonDecoder[ArticleData] = DeriveJsonDecoder.gen[ArticleData]
